include/seqan3/search/fm_index/bi_fm_index.hpp

```diff
@@ -222,7 +232,10 @@ class bi_fm_index
             throw std::invalid_argument("The text that is indexed cannot be empty.");
 
         this->text = &text;
-        rev_text = view::reverse(text);
+        if constexpr(is_collection)
+            rev_text = text | view::deep{view::reverse} | view::deep{view::persist} | view::reverse;
```

Comment:
could you not reuse the declared type from above to omit duplicate code?

Reply:
Mhm, I'd like to, but I didn't find any way to do it.

I need something like 
```cpp
static auto constexpr rev_text_view = std::conditional<is_collection, view::deep{view::reverse} | view::reverse, view::reverse>;
using rev_text_type = decltype (*text | rev_text_type);
```
But in a working version.

Reply:
Well yes, it is not so easy I guess, because you cannot propagate the object throught the views after you instantiated it. But in any case you could make a little detail helper function that makes the transformation. This can be reused to declare the type and to define the reverse text in the function